src/test/java/com/amihaiemil/camel/ScalarTest.java

```diff
@@ -102,11 +102,17 @@ public void equalsAndHashCode() {
             val
         );
 
+        final Scalar<String> nullScalar = new Scalar<String>(
+            Arrays.asList(Mockito.mock(AbstractNode.class)),
+            null
+        );
+
         MatcherAssert.assertThat(firstScalar, Matchers.equalTo(secondScalar));
         MatcherAssert.assertThat(secondScalar, Matchers.equalTo(firstScalar));
+
         MatcherAssert.assertThat(firstScalar, Matchers.equalTo(firstScalar));
-        MatcherAssert.assertThat(firstScalar, Matchers.equalTo(null));
-        MatcherAssert.assertThat(firstScalar.getClass() != secondScalar.getClass(), is(false));
+        MatcherAssert.assertThat(firstScalar, Matchers.notNullValue());
```

Comment:
@salikjan This assert verifies the same thing, that ``firstScalar`` is not ``null`` - but it does not call the equals method, as you can see in the coveralls report:

![image](https://cloud.githubusercontent.com/assets/6305156/21766567/f6d19228-d677-11e6-8a70-02b51ce90bbc.png)


The equivalent of this assert with simple Assert is:
``assertTrue(firstScalar != null)``, while what we want is ``assertTrue(!firstScalar.equals(null))``

This is the correct assert with matchers:

``MatcherAssert.assertThat(firstScalar, Matchers.not(Matchers.equalTo(null)));``